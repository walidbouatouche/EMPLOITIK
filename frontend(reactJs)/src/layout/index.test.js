import React from 'react'
import Layout from './index'
import Footer from '../compenents/footer'
import Menu from '../compenents/menu'
import { Helmet } from "react-helmet"
import toJson from 'enzyme-to-json';
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
Enzyme.configure({ adapter: new Adapter() })
const wrapper = shallow(<Layout />)
const title = 'Emploitik'

it('Layout render  correctly ', () => {

  expect(toJson(wrapper)).toMatchSnapshot();


});

it(' have only one footer ', () => {

  expect(wrapper.find(Footer).length).toEqual(1)


});

it('have only one Menu ', () => {
  expect(wrapper.find(Menu).length).toEqual(1)
})
it('have only one Helma with one title  ', () => {
  const wrapperHelmet = wrapper.find(Helmet);
  expect(wrapperHelmet.length).toEqual(1)
  expect(wrapperHelmet.find('title').length).toEqual(1)
 


})
