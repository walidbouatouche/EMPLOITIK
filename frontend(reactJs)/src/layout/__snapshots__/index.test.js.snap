// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout render  correctly  1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Emploitik
    </title>
  </HelmetWrapper>
  <Navabar />
  <div
    className="w3-row"
  >
    <div
      className="w3-col   w3-margin-top"
    >
      <br />
    </div>
  </div>
  <Footer />
</div>
`;
